MekHQ/src/mekhq/campaign/io/CampaignXmlParser.java

```diff
@@ -1370,9 +1370,16 @@ private static void postProcessParts(Campaign retVal, Version version) {
 
             Unit u = prt.getUnit();
             if (u != null) {
-                // get rid of any equipmentparts without locations or mounteds
+                // get rid of any equipmentparts without types, locations or mounteds
                 if (prt instanceof EquipmentPart) {
-                    EquipmentPart ePart = (EquipmentPart) prt;
+                    final EquipmentPart ePart = (EquipmentPart) prt;
+
+                    // Null Type... parsing failure
+                    if (ePart.getType() == null) {
```

Comment:
Let's log something about this if it happens.

Reply:
I looked at it when doing this. There's not enough data to make a useful log... and the useful data is previously logged.

Reply:
May need to move the check/log earlier in the call stack to catch the part type that was bad.

Reply:
We already log it when we fail to restore the equipment type. Just found it can be null at this point and... it's better to remove than deal with it down the line

Reply:
Is that good @NickAragua? Would like this done for 0.49.3 if possible

Reply:
Might as well. Not like that piece of equipment is recoverable anyway.